hw/ip/i2c/rtl/i2c_fsm.sv

```
@@ -286,33 +293,22 @@ module i2c_fsm (
   end
 
   // Bit counter on the target side
-  assign bit_ack = (bit_idx == 4'd9) && !start_det; // ack
+  assign bit_ack = (bit_idx == 4'd8) && !start_det; // ack
 
   // Increment counter on negative SCL edge
   always_ff @ (posedge clk_i or negedge rst_ni) begin : tgt_bit_counter
     if (!rst_ni) begin
       bit_idx <= 4'd0;
     end else if (start_det) begin
       bit_idx <= 4'd0;
-    end else if (scl_i_q && !scl_i) begin
+    end else if (!scl_i_q && scl_i) begin
```

It looks like this causes bit transitions while scl is high. We aren't allowed to do that (and DV should report an error here).

For the sake of choosing which bit to output, it seems like the falling edge of scl was likely right (modulo any "hold" time needed...).

---

ugh..let me just revert..